### PR TITLE
include: fix 'no implicit conversion of nil to String'

### PR DESCRIPTION
**Pull request by parkr**
This is when either 'dir' or 'file' is `nil` and passed `File.join`.

<details>
<summary>Backtrace</summary>
<pre>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/tags/include.rb#L115" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/tags/include.rb:115:in `join&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/tags/include.rb#L115" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/tags/include.rb:115:in `block in locate_include_file&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/tags/include.rb#L114" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/tags/include.rb:114:in `each&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/tags/include.rb#L114" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/tags/include.rb:114:in `locate_include_file&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/tags/include.rb#L129" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/tags/include.rb:129:in `render&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/block.rb#L151" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/block.rb:151:in `render_token&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/profiler/hooks.rb#L5" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/profiler/hooks.rb:5:in `block in render_token_with_profiling&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/profiler.rb#L80" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/profiler.rb:80:in `profile_token_render&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/profiler/hooks.rb#L4" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/profiler/hooks.rb:4:in `render_token_with_profiling&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/block.rb#L135" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/block.rb:135:in `block in render_all&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/block.rb#L122" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/block.rb:122:in `each&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/block.rb#L122" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/block.rb:122:in `render_all&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/block.rb#L108" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/block.rb:108:in `render&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/template.rb#L210" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/template.rb:210:in `block in render&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/template.rb#L262" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/template.rb:262:in `with_profiling&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/template.rb#L209" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/template.rb:209:in `render&#39;</a>
<a href="https://github.com/Shopify/liquid/blob/v3.0.6/lib/liquid/template.rb#L222" target="_blank">.bundle/ruby/2.3.0/gems/liquid-3.0.6/lib/liquid/template.rb:222:in `render!&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/liquid_renderer/file.rb#L28" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/liquid_renderer/file.rb:28:in `block (2 levels) in render!&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/liquid_renderer/file.rb#L40" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/liquid_renderer/file.rb:40:in `measure_bytes&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/liquid_renderer/file.rb#L27" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/liquid_renderer/file.rb:27:in `block in render!&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/liquid_renderer/file.rb#L47" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/liquid_renderer/file.rb:47:in `measure_time&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/liquid_renderer/file.rb#L26" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/liquid_renderer/file.rb:26:in `render!&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/renderer.rb#L134" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/renderer.rb:134:in `render_liquid&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/renderer.rb#L82" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/renderer.rb:82:in `run&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/site.rb#L463" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/site.rb:463:in `block in render_pages&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/site.rb#L461" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/site.rb:461:in `each&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/site.rb#L461" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/site.rb:461:in `render_pages&#39;</a>
<a href="https://github.com/jekyll/jekyll/blob/v3.3.1/lib/jekyll/site.rb#L191" target="_blank">.bundle/ruby/2.3.0/gems/jekyll-3.3.1/lib/jekyll/site.rb:191:in `render&#39;</a>
</pre>
</details>

### CODE REVIEW

File: lib/jekyll/tags/include.rb [hunk 1]
```diff
@@ -163,7 +163,7 @@ def load_cached_partial(path, context)
       end
 
       def valid_include_file?(path, dir, safe)
-        !(outside_site_source?(path, dir, safe) || !File.exist?(path))
+        !(outside_site_source?(path, dir, safe) || !File.file?(path))
```
Comment (parkr) [inline lib/jekyll/tags/include.rb]: We don't want this to be a directory.

Comment (ashmaroli) [inline lib/jekyll/tags/include.rb]: why not `!File.directory?(path)` ?

Comment (parkr) [inline lib/jekyll/tags/include.rb]: I believe checking to see if it's a file is easier to read - affirmations are easier for us to grok than negations.

Comment (ashmaroli) [inline lib/jekyll/tags/include.rb]: but its `!File.file?(path)` instead of `File.file?(path)`...

Comment (benbalter) [inline lib/jekyll/tags/include.rb]: yeah, If anything I'd write this as:

```ruby
!outside_site_source?(path, dir, safe) && File.file?(path)
```

Comment (parkr) [inline lib/jekyll/tags/include.rb]: Works for me! Updated to Ben's suggestion.

**Comment by parkr** [comment]
@jekyllbot: merge +bug

**Comment by ashmaroli** [comment]
Are we releasing a v3.3.2?  Thought the `master` already had changes that bumped the MINOR version..

**Pull request merged**